internal/runtime/client/client.go

```diff
@@ -382,6 +386,23 @@ func (c *client) CallExtension(ctx context.Context, hook runtimecatalog.Hook, fo
 	return nil
 }
 
+// prepareRequest creates a new request object by adding the appropriate values from the registration.
+func prepareRequest(request runtimehooksv1.RequestObject, registration *runtimeregistry.ExtensionRegistration) runtimehooksv1.RequestObject {
+	// Merge the settings from registration with the settings in the request.
+	// The values in request take precedence over the values in the registration.
+	// Create a deepcopy object to avoid side-effects on the request object.
+	request = request.DeepCopyObject().(runtimehooksv1.RequestObject)
+	settings := registration.Settings
+	if settings == nil {
+		settings = map[string]string{}
+	}
+	for k, v := range request.GetSettings() {
+		settings[k] = v
+	}
+	request.SetSettings(settings)
+	return request
```

Comment:
Maybe let's not also return the request. Modifying the input parameter + returning is always a bit confusing (as it's unclear if the input parameter is modified for the consumer). So I would only do it if it's necessary for the way a func is used

Reply:
Here the input request is not modified (by design to avoid side effects when a user calls the CallExtension function). Therefore we have to return the request object here.